lib/options.nix

```diff
@@ -466,6 +467,57 @@ rec {
       ${concatMapStringsSep "\n" (defFile: "  - ${defFile}") opt.files}
     '';
 
+  /**
+    Filters an options tree.
```

Comment:
This is very minimal. It's good to have "tree", but what does it do? How is the callback used?

Something like:

Filters an option tree, such that the returned tree only contains the options for which the callback returns `true`.

Arguments passed to the callback are:
- The option path, as a list of strings representing the attribute names.
- The option declaration at that path.

It is not possible to prune entire branches of the option tree in one go, as the callback is only invoked for each option (leaf).

------

Question: doesn't this leave empty branches behind?

Reply:
Applied.

Reply:
There's a test for leaving no empty branches now.